kernel-cert-script

```diff
@@ -0,0 +1,86 @@
+#!/bin/sh
+
+op=$(echo "$0" | sed -e "s/.*-//")
+
+while true ; do
+    case $1 in
+	--kmp)
+	    kmp="$2"
+	    shift 2
+	    ;;
+	--certs)
+	    certs="$2"
+	    shift 2
+	    ;;
+	*) break
+	    ;;
+    esac
+done
+
+run_mokutil () {
+    [ -z "$KERNEL_PACKAGE_SCRIPT_DEBUG" ] || echo mokutil "$@" > /dev/kmsg
+    mokutil "$@"
+}
+
+[ -z "$KERNEL_PACKAGE_SCRIPT_DEBUG" ] || echo "cert $op" kmp: "$kmp" certs: "$certs" > /dev/kmsg
+
+mok_rc=0
+
+case $op in
+    pre)
+	;;
+    preun)
+	if command -v mokutil >/dev/null; then
+	    for cert in "$certs"; do
+		cert="/etc/uefi/certs/${cert}.crt"
+		ln "$cert" "$cert.delete" || mok_rc=$?
```

Comment:
Please add a comment here ("preserve cert so that we can run in through mokutil in postun").

Btw why are we doing it like this? What's wrong with calling `mokutil --delete` in `%pre` ?

Reply:
The cert is shared among many kernels. All of them on most systems that don't have custom kernels nor PTFs.

So you link the certificate here to queue it for deletion, and then in postun you delete it if the certificate no longer exists - last kernel using it was removed.

Reply:
Also see https://github.com/hramrach/suse-module-tools for latest and possibly greatest revision of this patch.